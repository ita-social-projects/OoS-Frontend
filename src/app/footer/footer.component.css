:host {
  background-color: #ebebeb;
}
:host .trial-message {
  font-size: 0.9rem;
  font-family: "Innerspace";
  text-align: center;
  margin-top: 1.5rem;
}

.footer {
  padding: 2.375rem 5rem;
  bottom: 0;
  width: 100%;
}
.footer .min-logo {
  height: 41px;
}
@media (max-width: 1000px) {
  .footer {
    padding: 1.5rem 3rem;
    flex-direction: column !important;
    align-items: flex-start !important;
    row-gap: 2rem;
  }
  .footer > div {
    justify-content: space-between !important;
    -moz-column-gap: 2rem;
         column-gap: 2rem;
    width: 100%;
  }
}
@media (max-width: 900px) {
  .footer .min-logo {
    height: 33px;
  }
  .footer .company-logo {
    height: 15px;
  }
}
@media (max-width: 599px) {
  .footer {
    padding: 1rem;
    position: static;
    transform: none;
    row-gap: 1rem;
  }
  .footer > div {
    -moz-column-gap: 1rem;
         column-gap: 1rem;
    row-gap: 1rem;
  }
  .footer .min-logo {
    height: 25px;
  }
  .footer .company-logo {
    height: 12px;
  }
}/*# sourceMappingURL=footer.component.css.map */