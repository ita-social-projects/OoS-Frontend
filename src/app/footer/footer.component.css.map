{"version":3,"sources":["footer.component.scss","footer.component.css"],"names":[],"mappings":"AACA;EACE,yBAAA;ACAF;ADEE;EACE,iBAAA;EACA,yBAAA;EACA,kBAAA;EAEA,kBAAA;ACDJ;;ADKA;EAGE,sBAAA;EACA,SAAA;EACA,WAAA;ACJF;ADME;EACE,YAAA;ACJJ;ADOE;EAXF;IAYI,oBAAA;IACA,iCAAA;IACA,kCAAA;IACA,aAAA;ECJF;EDME;IACE,yCAAA;IACA,qBAAA;SAAA,gBAAA;IACA,WAAA;ECJJ;AACF;ADOE;EACE;IACE,YAAA;ECLJ;EDOE;IACE,YAAA;ECLJ;AACF;ADQE;EAjCF;IAkCI,aAAA;IACA,gBAAA;IACA,eAAA;IACA,aAAA;ECLF;EDOE;IACE,qBAAA;SAAA,gBAAA;IACA,aAAA;ECLJ;EDOE;IACE,YAAA;ECLJ;EDOE;IACE,YAAA;ECLJ;AACF","file":"footer.component.css"}