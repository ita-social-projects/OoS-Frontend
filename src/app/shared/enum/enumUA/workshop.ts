export enum PayRateTypeUkr {
  Classes = 'Заняття',
  Month = 'Місяць',
  Day = 'День',
  Year = 'Рік',
  Hour = 'Година',
  Course = 'Курс',
  AllPeriod = 'Увесь період'
}

export enum RecruitmentStatusUkr {
  RecruitmentOngoing = 'Набір триває',
  RecruitmentStoped = 'Набір призупинено',
  Stop = 'Призупинити набір',
  Open = 'Відкрити набір'
}

export enum NotificationWorkshopStatusUkr {
  Open = 'відкрито',
  Closed = 'закрито'
}