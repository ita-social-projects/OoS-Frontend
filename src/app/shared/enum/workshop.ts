export enum PayRateType {
  Classes = 1,
  Month,
  Day,
  Year,
  Hour,
  Course,
  AllPeriod
}

export enum WorkshopOpenStatus {
  Open = 'Open',
  Closed = 'Closed'
}

export enum DetailsTabTitlesParams {
  'AboutWorkshop',
  'AboutProvider',
  'Teachers',
  'OtherWorkshops',
  'Reviews',
  'Achievements'
}