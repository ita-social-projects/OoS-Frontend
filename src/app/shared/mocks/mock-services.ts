import { Observable } from 'rxjs';

export const MockOidcSecurityService = {
  authorize: (): void => {},
  logoff: (): void => {},
  getToken: (): void => {
    'some_token_eVbnasdQ324';
  },
  getAccessToken: (): Observable<any> =>
    new Observable((subscriber) => {
      subscriber.next(true);
      subscriber.complete();
    }),
  checkAuth: (): Observable<any> =>
    new Observable((subscriber) => {
      setTimeout(() => {
        subscriber.next(true);
        subscriber.complete();
      }, 1);
    })
};
