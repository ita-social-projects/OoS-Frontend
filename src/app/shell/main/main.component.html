<div class="main">
  <ng-container *ngIf="!(isLoadingData$ | async)">
    <div class="main-header">
      <h3>Найпопулярніші напрямки</h3>
      <a routerLink="/all-categories" class="main_link">
        {{ isMobile ? 'Всі' : 'Переглянути всі' }} <span class="material-icons arrow">arrow_forward</span>
      </a>
    </div>

    <ng-container *ngIf="(topDirections$ | async)?.length; then topDirections; else emptyData"></ng-container>

    <ng-template #topDirections>
      <div class="card-wrapper">
        <app-category-card
          class="card"
          *ngFor="let direction of topDirections$ | async"
          [direction]="direction.direction"
          [workshopsCount]="direction.workshopsCount">
        </app-category-card>
      </div>
    </ng-template>

    <div class="main-header">
      <h3>Найпопулярніші гуртки</h3>
      <a routerLink="/result" class="main_link">
        {{ isMobile ? 'Всі' : 'Переглянути всі' }} <span class="material-icons arrow">arrow_forward</span>
      </a>
    </div>

    <ng-container *ngIf="(topWorkshops$ | async)?.length; then topWorkshops; else emptyData"></ng-container>
  </ng-container>
  <ng-template #topWorkshops>
    <div class="card-wrapper">
      <app-workshop-card
        class="card"
        *ngFor="let workshop of topWorkshops$ | async"
        [workshop]="workshop"
        [isMainPage]="true"
        [userRoleView]="Role.unauthorized">
      </app-workshop-card>
    </div>
  </ng-template>

  <ng-template #emptyData>
    <div class="empty-list-wrapper">
      <p class="text">За результатами нічого не знайдено</p>
    </div>
  </ng-template>
</div>

<div *ngIf="(role$ | async) === Role.unauthorized" class="main_registration" fxLayoutAlign="space-between center">
  <div class="main_description">
    <h2 class="main_question">Надаєш освітні послуги?</h2>
    <h4 class="main_hint">Зареєструйся - допоможемо залучити нових учнів!</h4>
  </div>
  <div class="main_button-wrapper">
    <button mat-stroked-button class="main_button" (click)="register()">ЗАРЕЄСТРУВАТИСЯ</button>
  </div>
</div>
