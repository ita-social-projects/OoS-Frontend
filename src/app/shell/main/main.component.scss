@import "src/app/shared/styles/list-wrappers.scss";
@import "src/app/shared/styles/card-wrapper.scss";

.main {
  width: 100%;

  &_registration {
    padding: 32px 40px;
    background: #3849f9;
    border-radius: 5px;
    position: relative;
    height: 125px;
    margin-top: 4rem;
    &::before {
      content: "";
      display: block;
      position: absolute;
      background-image: url(../../../assets/images/half_circle.png);
      height: 100%;
      width: 62px;
      left: 0;
      top: 0;
    }
  }
  &_button {
    border-radius: 20px;
    padding: 10px 30px;
    font-size: 13px;
    line-height: 13px;
    border-color: white;
    color: white;
  }
  &_question {
    color: #ffffff;
  }
  &_hint {
    margin-top: 16px;
    color: #ffffff;
  }
  &_link {
    display: flex;
    font-weight: bold;
    color: #3849f9;
    align-items: center;
    .arrow {
      margin-left: 8px;
      font-size: 15px;
      line-height: 24px;
    }
  }
  &-header {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    margin: 1rem 0;
  }
 }

app-footer {
  visibility: hidden;
  position: absolute;
}

@media (max-width: 675px) {
  .main_registration {
    padding: 20px;
    height: inherit;
    display: block !important;

    .main_button-wrapper {
      text-align: center;
      margin-top: 10px;
    }
  }
}

@media (max-width: 750px) {
  app-footer {
    visibility: visible;
    position: static;
  }
}

@media (max-width: 415px) {
  .main_title {
    font-size: 16px !important;
  }
}
