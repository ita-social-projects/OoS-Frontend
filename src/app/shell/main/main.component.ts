import { Constants } from './../../shared/constants/constants';
import { Component, ElementRef, OnInit, ViewChild } from '@angular/core';
import { Actions, ofAction, Select, Store } from '@ngxs/store';
import { Observable, Subject } from 'rxjs';
import { SetCity, GetTopWorkshops } from 'src/app/shared/store/filter.actions';
import { FilterState } from 'src/app/shared/store/filter.state';
import { RegistrationState } from '../../shared/store/registration.state';
import { Direction } from 'src/app/shared/models/category.model';
import { MetaDataState } from 'src/app/shared/store/meta-data.state';
import { Workshop, WorkshopCard } from '../../shared/models/workshop.model';
import { GetDirections, GetTopDirections } from 'src/app/shared/store/meta-data.actions';
import { count, debounceTime, distinctUntilChanged, reduce, scan, takeUntil, tap, map } from 'rxjs/operators';
import { GetFilteredWorkshops } from './../../shared/store/filter.actions';


@Component({
  selector: 'app-main',
  templateUrl: './main.component.html',
  styleUrls: ['./main.component.scss'],

})

export class MainComponent implements OnInit {
  @Select(FilterState.topWorkshops)
  topWorkshops$: Observable<WorkshopCard[]>;
  @Select(RegistrationState.isAuthorized)
  isAuthorized$: Observable<boolean>;
  @Select(RegistrationState.parent)
  isParent$: Observable<boolean>;
  @Select(MetaDataState.topDirections)
  topDirections$: Observable<Direction[]>;
  destroy$: Subject<boolean> = new Subject<boolean>();
  @ViewChild('WorkshopsWrap') WorkshopsWrap: ElementRef;
  public parent: boolean;


  constructor(
    private store: Store,
    private actions$: Actions,
  ) { }


  ngOnInit(): void {
    this.store.dispatch([
      new GetTopDirections(),
      new GetTopWorkshops(Constants.WORKSHOPS_PER_PAGE)
    ]);

    this.actions$.pipe(ofAction(SetCity))
      .pipe(
        debounceTime(500),
        distinctUntilChanged(),
        takeUntil(this.destroy$))
      .subscribe(() => this.store.dispatch(new GetTopWorkshops(Constants.WORKSHOPS_PER_PAGE)));

    this.isParent$
      .pipe(
        takeUntil(this.destroy$))
      .subscribe(parent => this.parent = parent);
  }

  ngOnDestroy(): void {
    this.destroy$.next(true);
    this.destroy$.unsubscribe();
  }
  emptyWorkshops(): Array<Workshop> {
    let amountCardsInRow = 0;
    let workshops = [];
    let amountWorkshops = 0;
    this.topWorkshops$.pipe(map(x => workshops.push(x))).subscribe();
    amountWorkshops = workshops[0].length;
    if (this.WorkshopsWrap) {
      amountCardsInRow = Math.floor(Number((this.WorkshopsWrap.nativeElement.clientWidth) / 350));
    }
    let emptyWorkshops = (amountCardsInRow - amountWorkshops % amountCardsInRow) !== amountCardsInRow ? (amountCardsInRow - amountWorkshops % amountCardsInRow) : 0;
    return new Array(emptyWorkshops | 0);
  }
}
