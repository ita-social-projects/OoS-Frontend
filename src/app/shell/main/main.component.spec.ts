import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MainComponent } from './main.component';
import { NgxsModule } from '@ngxs/store';
import { OidcSecurityService } from 'angular-auth-oidc-client';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { MatSnackBarModule } from '@angular/material/snack-bar';
import { MockOidcSecurityService } from '../../shared/mocks/mock-services';
import { Component, Input } from '@angular/core';
import { Direction } from '../../shared/models/category.model';
import { RouterTestingModule } from '@angular/router/testing';
import { Workshop } from '../../shared/models/workshop.model';
import { FooterComponent } from 'src/app/footer/footer.component';

describe('MainComponent', () => {
  let component: MainComponent;
  let fixture: ComponentFixture<MainComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        MatSnackBarModule,
        NgxsModule.forRoot([]),
        HttpClientTestingModule,
        RouterTestingModule,
      ],
      declarations: [
        MainComponent,
        FooterComponent,
        MockMainCategoryCardComponent,
        MockMainWorkshopCardComponent
      ],
      providers: [
        { provide: OidcSecurityService, useValue: MockOidcSecurityService }
      ]
    })
      .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(MainComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});

@Component({
  selector: 'app-category-card',
  template: ''
})
class MockMainCategoryCardComponent {
  @Input() direction: Direction;
  @Input() workshopsCount: number;
  @Input() icons: {};
}

@Component({
  selector: 'app-workshop-card',
  template: ''
})
class MockMainWorkshopCardComponent {
  @Input() workshop: Workshop;
  @Input() isMainPage: boolean;
  @Input() userRole: string;
  @Input() parent: boolean;
}
