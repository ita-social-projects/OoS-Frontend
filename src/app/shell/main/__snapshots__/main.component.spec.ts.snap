// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent User is not authorized renders category cards with data from store 1`] = `
<div
  class="main_categories"
>
  <div
    class="main_header"
    fxlayoutalign="space-between center"
  >
    <h3>
      Найпопулярніші напрямки
    </h3>
    <a
      class="main_link"
      routerlink="/result"
    >
       Переглянути всі 
      <span
        class="material-icons arrow"
      >
        arrow_forward
      </span>
    </a>
  </div>
  <div
    class="categories-wrapper"
    fxlayout="row"
    fxlayoutalign="space-between"
    fxlayoutgap="1.5rem grid"
  >
    
    <app-category-card>
      <mat-card
        class="category"
      >
        <img
          alt="Category card title"
          class="icon"
          src="iconUrl"
        />
        <div
          class="details"
        >
          <h3
            class="title"
          >
            Category card title
          </h3>
          <div
            class="link"
          >
            <a
              routerlink="/result"
            >
               254 гуртки 
              <span
                class="material-icons arrow"
              >
                arrow_forward
              </span>
            </a>
          </div>
        </div>
      </mat-card>
    </app-category-card>
  </div>
</div>
`;

exports[`MainComponent User is not authorized renders org cards with data from store 1`] = `
<div
  class="main_cards"
>
  <div
    class="main_header"
    fxlayoutalign="space-between center"
  >
    <h3>
      Найпопулярніші гуртки
    </h3>
    <a
      class="main_link"
      routerlink="/result"
    >
       Переглянути всі 
      <span
        class="material-icons arrow"
      >
        arrow_forward
      </span>
    </a>
  </div>
  <div
    class="cards"
    fxlayout="row wrap"
    fxlayoutalign="space-between center"
  >
    
    
    <app-organization-card>
      <div
        class="card"
      >
        <div
          class="card_header"
        >
          <img
            alt="A,B"
            class="card_image"
            src="assets/icons/A,B_bg.png"
          />
          <button
            class="card_favorite"
          >
            <img
              alt="favorite"
              src="assets/icons/favorite.png"
            />
          </button>
          <a
            class="card_category"
            href="#"
          >
            <img
              alt="A,B"
              class="card_category-img"
              src="assets/icons/A,B.png"
            />
          </a>
        </div>
        <div
          class="card_body"
        >
          <h3
            class="card_title"
          >
             Org card title 
          </h3>
          <ul
            class="card_data"
          >
            <li
              class="card_data-item"
            >
              <img
                alt="rate"
                class="card_data-icon"
                src="assets/icons/rate.png"
              />
               5 
              <span
                class="card_rate-span"
              >
                (10)
              </span>
            </li>
            <li
              class="card_data-item"
            >
              <img
                alt="name"
                class="card_data-icon"
                src="assets/icons/name.png"
              />
              <a
                class="card_owner"
                href="#"
              >
                 owner 
              </a>
            </li>
            <li
              class="card_data-item"
            >
              <img
                alt="age"
                class="card_data-icon"
                src="assets/icons/age.png"
              />
               5-8 
            </li>
            <li
              class="card_data-item"
            >
              <img
                alt="price"
                class="card_data-icon"
                src="assets/icons/price.png"
              />
               20 
            </li>
            <li
              class="card_data-item"
            >
              <img
                alt="address"
                class="card_data-icon"
                src="assets/icons/address.png"
              />
               address 
            </li>
            <div
              class="card_footer"
            >
              <a
                class="card_link"
                href="#"
              >
                 Детальніше 
                <span
                  class="material-icons arrow"
                >
                  arrow_forward
                </span>
              </a>
            </div>
          </ul>
        </div>
      </div>
    </app-organization-card>
  </div>
</div>
`;
